diff -r 8c343a783777 src/windows/native/java/io/WinNTFileSystem_md.c
--- a/src/windows/native/java/io/WinNTFileSystem_md.c	Tue Oct 08 11:13:19 2013 -0700
+++ b/src/windows/native/java/io/WinNTFileSystem_md.c	Tue Jan 14 12:36:18 2014 -0800
@@ -637,10 +637,17 @@
         return NULL;
     }
 
-    /* Remove trailing space chars from directory name */
+    /* Remove trailing space chars from directory name,
+     * unless the \\?\ notation is used
+     */
     len = (int)wcslen(search_path);
-    while (search_path[len-1] == ' ') {
-        len--;
+    if (!(len > 4 && search_path[0] == L'\\'
+        && search_path[1] == L'\\'
+        && search_path[2] == L'?'
+        && search_path[3] == L'\\')) {
+        while (search_path[len-1] == ' ') {
+            len--;
+        }
     }
     search_path[len] = 0;
 
