--- u	2015-04-07 17:08:02.000000000 -0700
+++ ./src/com/aerofs/sp/server/BifrostClient.java	2015-04-07 17:09:02.000000000 -0700
@@ -12,6 +12,7 @@
 import java.net.URL;
 import java.util.concurrent.ExecutionException;
 
+import static com.aerofs.base.config.ConfigurationProperties.getStringProperty;
 import static com.aerofs.lib.LibParam.Bifrost.INTERNAL_BIFROST_HOST;
 import static com.aerofs.lib.LibParam.Bifrost.INTERNAL_BIFROST_PORT;
 import static com.aerofs.sp.server.lib.SPParam.ZELDA_ID;
@@ -36,7 +37,7 @@
                         "&code_type=device_authorization&client_id=" + ZELDA_ID +
                         "&client_secret=" + ZELDA_SECRET + "&expires_in=" + expires;
 
-        URL bifrostBaseUrl = new URL("http", INTERNAL_BIFROST_HOST, INTERNAL_BIFROST_PORT, "");
+        URL bifrostBaseUrl = new URL("http", getStringProperty("hack_bifrost_service_ip", ""), INTERNAL_BIFROST_PORT, "");
         URL bifrostTokensUrl = new URL(bifrostBaseUrl, "/token");
 
         HttpURLConnection conn = (HttpURLConnection)bifrostTokensUrl.openConnection();
@@ -53,7 +54,7 @@
     public void deleteToken(String oldToken)
             throws IOException
     {
-        URL bifrostBaseUrl = new URL("http", INTERNAL_BIFROST_HOST, INTERNAL_BIFROST_PORT, "");
+        URL bifrostBaseUrl = new URL("http", getStringProperty("hack_bifrost_service_ip", ""), INTERNAL_BIFROST_PORT, "");
         URL bifrostTokensUrl = new URL(bifrostBaseUrl, "/token/" + oldToken);
 
         HttpURLConnection conn = (HttpURLConnection)bifrostTokensUrl.openConnection();
