--- web/views/maintenance/templates/device_restriction.mako	2015-01-09 10:31:25.000000000 -0800
+++ buildroot/opt/bunker/web/views/maintenance/templates/device_restriction.mako	2015-03-02 11:16:36.000000000 -0800
@@ -2,7 +2,7 @@
 <%! page_title = "Device Restriction" %>
 
 <%namespace name="csrf" file="csrf.mako"/>
-<%namespace name="bootstrap" file="bootstrap.mako"/>
+<%namespace name="loader" file="loader.mako"/>
 <%namespace name="spinner" file="spinner.mako"/>
 <%namespace name="progress_modal" file="progress_modal.mako"/>
 
@@ -167,7 +167,7 @@
 </%progress_modal:html>
 
 <%block name="scripts">
-    <%bootstrap:scripts/>
+    <%loader:scripts/>
     <%spinner:scripts/>
     <%progress_modal:scripts/>
     <script>
@@ -204,11 +204,12 @@
 
         function restartDeviceServices() {
             var $progress = $('#${progress_modal.id()}');
-            runBootstrapTask('restart-services-for-device-authorization', function() {
+            reboot('current', function() {
                 $progress.modal('hide');
                 showSuccessMessage('New configuration is saved.');
-            }, function() {
+            }, function(xhr) {
                 $progress.modal('hide');
+                showErrorMessageFromResponse(xhr);
             });
         }
 
@@ -229,11 +230,12 @@
 
         function restartMDMServices() {
             var $progress = $('#${progress_modal.id()}');
-            runBootstrapTask('restart-services-for-mdm', function() {
+            reboot('current', function() {
                 $progress.modal('hide');
                 showSuccessMessage('New configuration is saved.');
-            }, function() {
+            }, function(xhr) {
                 $progress.modal('hide');
+                showErrorMessageFromResponse(xhr);
             });
         }
 
